azerothcore/azerothcore-wotlk
# fix(Core/ZG): Multiple improvements to High Priestess Mar'li

## Changes Proposed:
-  Modernize her script.
-  Unhack spider spawning.
-  Fixed a lot of timers.
-  Added missing mechanics.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11531
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11529
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11617
- Partly closes https://github.com/azerothcore/azerothcore-wotlk/issues/11532 (spiders are not spawned anymore on spider phase)

## SOURCE:

Timers from CMangos/Provided source.

## Tests Performed:

- Tested in-game.

## How to Test the Changes:


1. Engage Mar'li and check she casts all the correct spells.
2. Wait for the spawns of Mar'li and check their growth.
3. With more players, check charge spell (this was not tested but the code was improved).

## Known Issues and TODO List:


- [ ] For some reason, gobjects don't despawn on activation.
- [ ] Gobjects casts... The spider spawn spells don't work as they should because of this, in consequence the spiders are spawning on Mar'li instead on the eggs.
- [x] Melee damage from Mar'li seems to be too high ?


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
